Sources/Bag.swift

```diff
@@ -22,11 +41,22 @@ public struct Bag<Element> {
 	/// - parameters:
 	///   - value: A value that will be inserted.
 	@discardableResult
-	public mutating func insert(_ value: Element) -> RemovalToken {
-		let token = RemovalToken()
-		let element = BagElement(value: value, token: token)
+	public mutating func insert(_ value: Element) -> Token {
+		var timestamp: UInt64 = 0
+
+		repeat {
+			// `DispatchTime` is expected to use monotonic clocks. That said a guard is
+			// put here just in case.
+			timestamp = DispatchTime.now().uptimeNanoseconds
+		} while timestamp <= lastTimestamp
+
+		lastTimestamp = timestamp
+
+		let token = Token(timestamp)
```

Comment:
Sure… but if we still used a `UInt64`, then this would have the same domain. Using a counter would start at an earlier value. It would also only increment by 1 for each insertion—instead of skipping values in the case of a timestamp. So using a counter would be less likely to overflow.